eclipse-collections/src/main/java/org/eclipse/collections/impl/lazy/FlatCollectIterable.java

```diff
@@ -78,6 +79,32 @@ public V detect(Predicate<? super V> predicate)
     }
 
     @Override
+    public Optional<V> detectOptional(Predicate<? super V> predicate)
+    {
+        V[] result = (V[]) new Object[1];
+        Iterate.anySatisfy(this.adapted, each -> {
+            Iterable<V> iterable = this.function.valueOf(each);
+            if (iterable == null)
+            {
+                throw new NullPointerException();
+            }
+            return Iterate.anySatisfy(iterable, each1 -> {
+                if (predicate.accept(each1))
+                {
+                    if (each == null)
+                    {
+                        throw new NullPointerException();
+                    }
+                    result[0] = each1;
```

Comment:
I don't understand checking `each == null` all the way down here. If we made it here, then `function.valueOf(each)` already returned a non-null, non-empty list. I think we should either check earlier in the outer lambda, or not at all.